### ISSUE TITLE
`quotes` not working with backticks and function string arguments

### ISSUE THREAD

**Issue by whitneyit** [eslint/eslint#5759]
**What version of ESLint are you using?**
eslint@2.5.3

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{
	"env": {
		"browser": true,
		"es6": true,
		"node": true
	},
	"extends": "eslint:recommended",
	"rules": {
		"quotes": [
			"error",
			"backtick",
			"avoid-escape"
		]
	}
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
const foo = `Lorem`;
const bar = `Ipsum ${foo.replace("L", "l")}`;
```

**What did you expect to happen?**
For the double quotes in the `replace` function to be valid because of the `avoid-escape` option.

**What actually happened? Please include the actual, raw output from ESLint.**
```
 /home/whitneyit/Code/test.js
  2:7   error  'bar' is defined but never used  no-unused-vars
  2:34  error  Strings must use backtick        quotes
  2:39  error  Strings must use backtick        quotes
```

Is this the intended behaviour? If not, would you welcome a PR?

**Comment by mysticatea** [comment]
Thank you for this issue.

You can use backtick in `replace` function arguments.

```js
const bar = `Ipsum ${foo.replace(`L`, `l`)}`;
```

**Comment by whitneyit** [comment]
I didn't even know that was possible :open_mouth:

With that info, does this mean that the warning is correct? I feel like it should throw a different error message so as to inform the user that is is possible to nest backticks.

I dunno, could just be that I'm silly :flushed:

**Comment by mysticatea** [comment]
Yes, the warning is correct.
We can write any expression (yeah, includes template strings) between `${` and `}`.

```js
const foo = `Hello, ${`World`}!`;
```

Surely, this is funny. :smile: 
But I'm not sure whether we need to change the error message or not.

Further Reading: https://tc39.github.io/ecma262/2016/#prod-TemplateLiteral

**Comment by platinumazure** [comment]
@mysticatea Just to confirm, you don't think this is a bug? So either this is invalid, or an enhancement request to change the error message?

If you don't think this is a bug, one of us can label this as an enhancement under evaluation. Thanks!

**Comment by IanVS** [comment]
I agree this isn't a bug, and I'm not really a fan of changing the error message.  Going down that rabbit hole can get very complicated very quickly, for little (if any) gain.  Even if there are some unexpected situations like this that cause some research or an issue to check whether it's a bug, in the end it all worked out and now there's a github issue that others can learn from as well.  So, personally I'm :-1: on any changes here.

**Comment by IanVS** [comment]
Closing this, but feel free to speak up if you think we should reopen.
